packages/api/src/router/user.ts

```diff
@@ -86,26 +86,34 @@ export const userRouter = router({
         where: { userId: ctx.user.id },
       });
 
-      const promises = accounts
-        .filter((account) => account.provider === "google")
-        .map((account) => {
-          return getGoogleUsers({
-            search: input.search,
-            accessToken: account.access_token,
-            refreshToken: account.refresh_token,
-          });
+      const promises = accounts.flatMap((account) => {
+        // if account is not google, skip
+        if (account.provider !== "google") return [];
+        // if no email is set for this account, skip
+        if (!account.email) return [];
+        // if account is not a workspace accoount, skip
+        if (/gmail.com|googlemail.com/.test(account.email || "")) return [];
```

Comment:
skip if the account is a personal gmail

nb: removed the `|| ""` in a following commit.